caffe2/CMakeLists.txt

```diff
@@ -1491,6 +1492,11 @@ if(USE_DISTRIBUTED)
   if(USE_TENSORPIPE)
     target_compile_definitions(torch_cpu PUBLIC USE_TENSORPIPE)
   endif()
+  # Pass USE_NVSHMEM to torch_cpu because availability determination needs to be
+  # generically callable from torch_cpu side
+  if(USE_NVSHMEM)
+    target_compile_definitions(torch_cpu PUBLIC USE_NVSHMEM)
```

Comment:
A bit confused why this part is necessary.

Reply:
The function `is_nvshmem_available()` is defined in `SymmetricMemory.cpp` which is compiled with torch_cpu instead of torch_cuda. And I think this function is compiled only once -- I don't know how to use a "later" compilation (say with torch_cuda) to override a "earlier" compilation (say with torch_cpu). Do you have a suggestion? Thanks!

Reply:
Removed